docs/schema-loader.md

```diff
@@ -1,50 +1,83 @@
 # ScalarDB Schema Loader
 
-ScalarDB has its own data model and schema, that maps to the implementation specific data model and schema.
-Also, it stores internal metadata (e.g., transaction ID, record version, transaction status) for managing transaction logs and statuses when you use the Consensus Commit transaction manager.
-It is a little hard for application developers to manage the schema mapping and metadata for transactions, so we offer a tool called ScalarDB Schema Loader for creating schema without requiring much knowledge about those.
+ScalarDB has its own data model and schema that maps to the implementation-specific data model and schema. In addition, ScalarDB stores internal metadata, such as transaction IDs, record versions, and transaction statuses, to manage transaction logs and statuses when you use the Consensus Commit transaction manager.
 
-There are two ways to specify general CLI options in Schema Loader:
-  - Pass a ScalarDB configuration file and database/storage-specific options additionally.
-  - Pass the options without a ScalarDB configuration (Deprecated).
+Since managing the schema mapping and metadata for transactions can be difficult, you can use ScalarDB Schema Loader, which is a tool to create schemas that doesn't require you to need in-depth knowledge about schema mapping or metadata.
 
-Note that this tool supports only basic options to create/delete/repair/alter a table. If you want
-to use the advanced features of a database, please alter your tables with a database specific tool after creating them with this tool.
+You have two options to specify general CLI options in Schema Loader:
 
-# Usage
+- Pass the ScalarDB properties file and database-specific or storage-specific options.
+- Pass database-specific or storage-specific options without the ScalarDB properties file. (Deprecated)
 
-## Install
+{% capture notice--info %}
+**Note**
 
-The release versions of `schema-loader` can be downloaded from [releases](https://github.com/scalar-labs/scalardb/releases) page of ScalarDB.
+This tool supports only basic options to create, delete, repair, or alter a table. If you want to use the advanced features of a database, you must alter your tables with a database-specific tool after creating the tables with this tool.
+{% endcapture %}
 
-## Build
+<div class="notice--info">{{ notice--info | markdownify }}</div>
+
+## Set up Schema Loader
+
+Select your preferred method to set up Schema Loader, and follow the instructions.
+
+<div id="tabset-1">
+<div class="tab">
+  <button class="tablinks" onclick="openTab(event, 'Install', 'tabset-1')" id="defaultOpen-1">Install</button>
+  <button class="tablinks" onclick="openTab(event, 'Build', 'tabset-1')">Build</button>
+  <button class="tablinks" onclick="openTab(event, 'Docker', 'tabset-1')">Docker</button>
```

Comment:
@josh-wong @feeblefakie 
I'm wondering if we really need the `Build` and `Docker` sections in this document. For the OSS documentation perspective, our target audience comprises both OSS developers and ScalarDB users, so including the `Build` and `Docker` sections might be beneficial for the developers. But for the product documentation perspective, I not sure we need the `Build` and `Docker` sections because ScalarDB users might not be interested in them.

Reply:
As @brfrn169 said, we can probably remove `Build` from the options since it's more like a user guide.
We also download/install `Docker`, so it is probably better to name them as follows?

* (Use) Fat JAR
* (Use) Docker container

Also, `Docker` should probably be more recommended in production environment, so I don't think we can say either is recommended?
What do you think? > @brfrn169

Reply:
> Also, Docker should probably be more recommended in production environment, so I don't think we can say either is recommended?

Actually, I'm not sure which `Fat jar` or `Docker container` should be used in production environments. Considering that we also have the SQL CLI (available only in the enterprise edition) as a tool similar to the Schema Loader, it seemed to me that a 'Fat jar' might be easier for users to use. If uses can always use `Fat jar` in any environments, we can say `Fat jar` is recommended or just remove `Docker container`.

@kota2and3kan, do you have any insights on this?

Reply:
@feeblefakie @brfrn169  

I think we can keep both `Fat Jar` and `Docker container` and we don't need to say `which is recommended`. I think users can select out of two or more methods.

If users use `Fat Jar`, from the network perspective, they have to allow connections to access ScalarDB Cluster from some servers like the bastion server that is used for management operations. However, in the production environment, it would be better to avoid allowing unnecessary connections (i.e., other than application connections). For this reason, it might be better to use `Docker container` instead of `Fat Jar`. This is because the `Docker container` will be deployed as a pod in the Kubernetes cluster so that basically they don't need to allow additional connections other than application connections.

From this perspective (security perspective), it might be better to use the `Docker container` instead of `Fat Jar` in the production environment. But, as I mentioned, I think it is an optional thing and users can choose one of them based on their systems requirement or ScalarDB usages.

Also, as @brfrn169 mentioned, especially for commercial license users, they can use SQL CLI to create schemas and tables. In addition, users can create schemas and tables by using Admin API from their applications. These are the additional options. However, this document is a `ScalarDB Schema Loader` document. So, I think we don't need to mention SQL CLI or Admin API here.

On the other hand, I agree that the `Fat Jar` is easy to use. If we use the `Docker container` (especially in the Kubernetes environment) we need to create several manifest fils (e.g., pod, config map that includes schema configuration, and secret that includes credentials). So, ideally, I think it would be better to provide Helm Chart to use the `Docker container` like `ScalarDL Schema Loader`.

In conclusion, I think the following steps might be better.

1. In this PR, we keep `Fat Jar` and `Docker container` without any `recommendation` mark so that users can know `there are several options` and can choose options based on their requirements.
2. Create a helm chart for ScalarDB Schema Loader to use it easily on the Kubernetes environment.
3. Update the `Docker container` to `Docker container with Helm Chart` in this document.

What do you think?

Reply:
@brfrn169  
Thank you for your reply!

> I basically agree that we should offer both options, allowing users to choose based on their requirements.

Yes. I agree too.

> But what are your thoughts on the above?

Ah, sorry. I mixed some tools in my descriptions.
I agree that creating a Helm Chart for SQL CLI is not easy. We don't need to provide it.

I think we can create a Helm Chart for **ScalarDB Schema Loader**.
My image is the following.

* Custom values file

  ```yaml
  scalardbSchemaLoader:

    image:
      # For ScalarDB Schema Loader
      repository: ghcr.io/scalar-labs/scalardb-schema-loader
      # For ScalarDB Cluster Schema Loader
      repository: ghcr.io/scalar-labs/scalardb-cluster-schema-loader
  
    databaseProperties:
      # For ScalarDB Schema Loader (ScalarDB Server)
      scalar.db.transaction_manager=grpc
      scalar.db.contact_points=172.20.255.200
      scalar.db.contact_port=60051

      # For ScalarDB Cluster Schema Loader
      scalar.db.transaction_manager=cluster
      scalar.db.contact_points=direct-kubernetes:default/scalardb-cluster-envoy
  
      # Direct access to backend DB (I think this is a rare case)
      scalar.db.storage=jdbc
      scalar.db.contact_points=jdbc:postgresql://postgresql-scalardb.default.svc.cluster.local:5432/postgres
      scalar.db.username=${env:SCALAR_DB_POSTGRES_USERNAME}
      scalar.db.password=${env:SCALAR_DB_POSTGRES_PASSWORD}

    schemaJson:
      {
          "s0.t0": {
              "transaction": true,
              "partition-key": [
                  "c1"
              ],
              "clustering-key": [
                  "c2"
              ],
              "columns": {
                  "c1": "INT",
                  "c2": "INT",
                  "c3": "TEXT"
              }
          }
      }
    
    commandArgs:
      - --coordinator
      - --no-backup
      - --no-scaling
      - --repair-all 
    
    # Provide credentials via env variables like "SCALAR_DB_POSTGRES_USERNAME".
    secretName: "scalardb-credentials-secret"

    # Provide token to pull ScalarDB Cluster's Schema Loader image.
    imagePullSecrets:
      - name: secret-for-scalar
  ```

* Deploy (create schema on Kubernetes environment)

  ```console
  $ helm install scalardb-schema-loader scalar-labs/scalardb-schema-loader -f custom-values.yaml
  ```

In this way, I think users can use `ScalarDB Schema Loader` easily in a production (especially on Kubernetes) environment without additional bastion servers or allowing network connections.

So, in the future, users can see the following options on this document.

* Fat Jar
* Docker container
* Helm Chart

---

Also, I think it would be better to list the options as follows if we mention SQL CLI in this document.

* Fat Jar
* Docker container
* Helm Chart
* SQL CLI (Enterprise license only) Note: This option only includes links to SQL CLI documents.

However, I think this document is a document of ScalarDB Schema Loader. So, I think it would be better to consider whether we should mix SQL CLI in this document or not.

I think it's OK to include SQL CLI if the purpose of this document is `how to load schema`.

However, I think it's not good to include SQL CLI if the purpose of this document is `how to use ScalarDB Schema Loader`.

What do you think?

Reply:
As discussed offline with @kota2and3kan, we concluded that we keep both `Fat Jar` and `Docker container`, and in the future we can add `Helm Chart` (maybe as enterprise only) in addition to them.

@josh-wong Can you please delete `Build` and keep `Fat Jar` and `Docker container`? And we don't need to say `which is recommended`.

Reply:
@brfrn169 @kota2and3kan Thank you for confirming which methods to keep! I've removed the `Build` content.

In the Docker section, we also mention the following. Can we keep this part or should we remove it?:

````markdown
You can also build the Docker image as follows by running the following command:

```console
$ ./gradlew schema-loader:docker
```
````

Reply:
@josh-wong Thank you for removing the `Build` content.

> In the Docker section, we also mention the following. Can we keep this part or should we remove it?:

I think we should remove this because it's likely not of interest to users. Thanks.

Reply:
Thank you for confirming! I've removed that part.